ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -173,109 +173,84 @@ private[server] object ServerHelpers extends ServerHelpersPlatform {
-                // protocol has not been agreed via handshake
-                runConnection(
-                  socket,
-                  logger,
-                  idleTimeout,
-                  receiveBufferSize,
-                  maxHeaderSize,
-                  requestHeaderReceiveTimeout,
-                  httpApp,
-                  errorHandler,
-                  onWriteFailure,
-                  createRequestVault,
-                  webSocketKey,
-                  ByteVector.empty,
-                  enableHttp2,
-                ).drain
-              case (socket, None) => // Cleartext Protocol
-                enableHttp2 match {
-                  case true =>
-                    // Http2 Prior Knowledge Check, if prelude is first bytes received tread as http2
-                    // Otherwise this is now http1
-                    Stream.eval(H2Server.checkConnectionPreface(socket)).flatMap {
-                      case Left(bv) =>
-                        runConnection(
-                          socket,
-                          logger,
-                          idleTimeout,
-                          receiveBufferSize,
-                          maxHeaderSize,
-                          requestHeaderReceiveTimeout,
-                          httpApp,
-                          errorHandler,
-                          onWriteFailure,
-                          createRequestVault,
-                          webSocketKey,
-                          bv, // Pass read bytes we thought might be the prelude
-                          enableHttp2,
-                        ).drain
-                      case Right(_) =>
-                        Stream
-                          .resource(
-                            H2Server.fromSocket[F](
-                              socket,
-                              httpApp,
-                              H2Frame.Settings.ConnectionSettings.default,
-                              logger,
-                            )
-                          )
-                          .drain
-                    }
-                  // Since its not enabled, run connection normally.
-                  case false =>
-                    runConnection(
-                      socket,
-                      logger,
-                      idleTimeout,
-                      receiveBufferSize,
-                      maxHeaderSize,
-                      requestHeaderReceiveTimeout,
-                      httpApp,
-                      errorHandler,
-                      onWriteFailure,
-                      createRequestVault,
-                      webSocketKey,
-                      ByteVector.empty,
-                      enableHttp2,
-                    ).drain
+
+    def upgradeSocket(socketInit: Socket[F]): Resource[F, (Socket[F], Option[String])] =
+      tlsInfoOpt match {
+        case None => Resource.pure(socketInit, None)
```

Comment:
fails on 2.12: https://github.com/http4s/http4s/actions/runs/3826329960/jobs/6510029546#step:10:105

Reply:
Woops, sorry, should have had more parens:
```suggestion
        case None => Resource.pure((socketInit, None))
```
tested locally, worked with 2.12.17